CMakeLists.txt

```diff
@@ -27,13 +27,19 @@ if(NOT CMAKE_INSTALL_PREFIX)
 endif()
 
 project(dawn C CXX)
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.14.5)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 set(YODA_ROOT CACHE PATH "path to yoda package")
 
 if("${YODA_ROOT}" STREQUAL "")
-  message(FATAL_ERROR "YODA_ROOT not found! Try specifying it in the environment via -DYODA_ROOT=<>")
+  include(FetchContent)
+  FetchContent_Declare(
```

Comment:
There is already a way to fetch yoda in here:
https://github.com/MeteoSwiss-APN/dawn/blob/master/bundle/cmake/DawnCloneRepository.cmake

Would it make sense to apply these changes there?